replication/client_lib/include/rrdb_client.h

```diff
@@ -0,0 +1,55 @@
+#pragma once
+
+#include <string>
+#include "rrdb_error.h"
+
+namespace dsn { namespace apps {
+
+// irrdb_client is the base class that users use to access a specific cluster.
+// user can't create an instance, instead, they should get an instace through client_factory::get_context(const char* cluster_name),
+// and destroy through client_factory::drop_context(client* context);
+//
+// the class of client provides the basic operation to:
+// set/get/delete the value of a key in a app.
+
+class irrdb_client
+{
+public:
+    // destructor
+    virtual ~irrdb_client(){}
+
+    virtual const char* get_cluster_name() const = 0;
+    virtual const char* get_app_name() const = 0;
+
+    virtual int set(
+        const std::string& hashkey,
```

Comment:
@imzhenyu finally I decide to leave the pr and let you decide whether or not merge to master or open a new branch. I updated file https://github.com/imzhenyu/rocksdb/blob/master/replication/src/rrdb.client.h.(add get_key_hash function) then this branch may not work with Microsoft/rDSN.
if you think it necessary to open a new branch, and left this one for Microsoft/rDSN. please help me create a new branch here. thanks a lot.